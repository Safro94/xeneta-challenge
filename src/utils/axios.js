import axios from 'axios';
import https from 'https';

const opts = {
	httpsAgent: new https.Agent({ rejectUnauthorized: false }),
	baseURL: process.env.REACT_APP_SERVER_URL,
};

const interceptorError = error => Promise.reject(error.response);

const instance = axios.create(opts);

instance.interceptors.response.use(response => response, interceptorError);
instance.defaults.headers.common['x-api-key'] = process.env.REACT_APP_API_KEY;

export default instance;
