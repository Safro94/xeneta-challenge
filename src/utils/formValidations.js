const validateSearchForm = ({
	departure,
	destination,
	departureDate,
	returnDate,
}) => {
	if (!departure || !destination || !departureDate) {
		return false;
	}

	if (departure.code === destination.code) {
		return false;
	}

	if (returnDate && returnDate < departureDate) {
		return false;
	}

	return true;
};

const errorCompare = (trip, message) => {
	const { departure, destination } = trip;

	const status = !departure && !destination ? true : false;

	return {
		message,
		status: !status && departure?.code === destination?.code,
	};
};

export { validateSearchForm, errorCompare };
