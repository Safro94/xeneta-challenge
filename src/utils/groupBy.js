import { marketPositions } from 'constants/marketPositions';

const groupBy = data =>
	data.reduce((accum, item) => {
		if (!accum[marketPositions.low]) accum[marketPositions.low] = [];
		if (!accum[marketPositions.average]) accum[marketPositions.average] = [];
		if (!accum[marketPositions.high]) accum[marketPositions.high] = [];

		accum[marketPositions.low].push({ x: item.day, y: item.low });
		accum[marketPositions.average].push({ x: item.day, y: item.mean });
		accum[marketPositions.high].push({ x: item.day, y: item.high });

		return accum;
	}, {});

export default groupBy;
