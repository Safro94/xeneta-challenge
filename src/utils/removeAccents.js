const removeAccents = value => {
	const map = {
		a: 'á|à|ã|â',
		e: 'é|è|ê',
		i: 'í|ì|î',
		o: 'ó|ò|ô|õ',
		u: 'ú|ù|û|ü',
	};

	for (let pattern in map) {
		value = value.replace(new RegExp(map[pattern], 'g'), pattern);
	}

	return value;
};

export { removeAccents };
