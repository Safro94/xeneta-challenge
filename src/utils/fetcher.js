import axios from 'utils/axios';

const fetcher = async ({ method = 'get', url, data = null }) => {
	return axios[method](url, JSON.parse(data)).then(
		response => response.data,
		error => {
			throw error;
		}
	);
};

export default fetcher;
