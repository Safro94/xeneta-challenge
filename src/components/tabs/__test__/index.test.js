import { render, screen } from 'utils/test-utils';
import Tabs from '../';

describe('Tabs', () => {
	const tabs = ['Tab', 'Tab2'];
	const selected = 'Tab';
	const setSelected = jest.fn();

	it('should match the snapshot', () => {
		const { container } = render(
			<Tabs tabs={tabs} selected={selected} setSelected={setSelected}>
				<Tabs.Tab isSelected={selected === 'Tab'}>
					<h1>Tab</h1>
				</Tabs.Tab>
				<Tabs.Tab isSelected={selected === 'Tab2'}>
					<h1>Tab2</h1>
				</Tabs.Tab>
			</Tabs>
		);
		expect(container.firstElementChild).toMatchSnapshot();
	});

	it('should have two buttons', () => {
		render(
			<Tabs tabs={tabs} selected={selected} setSelected={setSelected}>
				<Tabs.Tab isSelected={selected === 'Tab'}>
					<h1>Tab content</h1>
				</Tabs.Tab>
				<Tabs.Tab isSelected={selected === 'Tab2'}>
					<h1>Tab2 content</h1>
				</Tabs.Tab>
			</Tabs>
		);

		expect(screen.getByText('Tab')).toBeInTheDocument();
		expect(screen.getByText('Tab2')).toBeInTheDocument();
		expect(screen.getAllByRole('button')).toHaveLength(2);
	});

	it('should render the content', () => {
		render(
			<Tabs tabs={tabs} selected={selected} setSelected={setSelected}>
				<Tabs.Tab isSelected={selected === 'Tab'}>
					<h1>Tab content</h1>
				</Tabs.Tab>
			</Tabs>
		);

		expect(screen.getByText(/tab content/i)).toBeInTheDocument();
	});
});
