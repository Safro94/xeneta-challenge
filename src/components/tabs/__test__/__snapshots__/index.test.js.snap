// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should match the snapshot 1`] = `
<ul
  class="tabList"
>
  <li>
    <button
      class="tabLink tabLinkActive"
      type="button"
    >
      Tab
    </button>
  </li>
  <li>
    <button
      class="tabLink"
      type="button"
    >
      Tab2
    </button>
  </li>
</ul>
`;
